File: Codeforces/Round223-Div2/A/main.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scanln(&n)
	card := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &card[i])
	}
	var l, r int = 0, n - 1
	var a, b int = 0, 0
	for i := 0; i < n; i++ {
		var temp int = 0
		if card[l] > card[r] {
			temp = card[l]
			l++
		} else {
			temp = card[r]
			r--
		}
		if i&1 == 1 {
			b += temp
		} else {
			a += temp
		}
	}
	fmt.Println(a, b)
}

File: Codeforces/Round223-Div2/A/main.cpp
#include <map>
#include <set>
#include <list>
#include <cmath>
#include <queue>
#include <stack>
#include <bitset>
#include <vector>
#include <cstdio>
#include <string>
#include <climits>
#include <sstream>
#include <cstdlib>
#include <cstring>
#include <iostream>
#include <algorithm>
using namespace std;
#define PB push_back
#define MP make_pair
#define SZ(v) ((int)(v).size())
#define abs(x) ((x) > 0 ? (x) : -(x))
#define FOREACH(e,x) for(__typeof(x.begin()) e=x.begin();e!=x.end();++e)
typedef long long LL;

const int MAXN = 1000 + 5;
int n;
int card[MAXN];

int main() {
	cin >> n;
	for (int i = 0; i < n; i++)
		cin >> card[i];
	int l = 0, r = n - 1;
	int a = 0, b = 0;
	for (int i = 0; i < n; i++) {
		int temp = 0;
		if (card[l] > card[r]) {
			temp = card[l++];
		} else {
			temp = card[r--];
		}
		if (i & 1) b += temp;
		else a += temp;
	}
	cout << a << " " << b << endl;
	return 0;
}

